crates/binstalk/src/helpers/git.rs

```diff
@@ -53,7 +74,7 @@ impl FromStr for GitUrl {
 }
 
 #[derive(Debug)]
-pub struct Repository(gix::Repository);
+pub struct Repository(pub(crate) gix::Repository);
```

Comment:
If you accept my proposal of adding `get_head_commit_entry_data_by_path`, then we could change this to:

```suggestion
pub struct Repository(gix::ThreadSafeRepository);
```